gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/store/GemFireStore.java

```diff
@@ -2308,6 +2299,15 @@ public void setDBName(String dbname) {
         this.database.setdisableStatementOptimizationToGenericPlan();
       }
     }
+    // install the GemFireXD specific thread dump signal (URG) handler
+    if (!isSnappyStore()) {
+      try {
+        SigThreadDumpHandler.install();
+      } catch (Throwable t) {
+        SanityManager.DEBUG_PRINT("fine:TRACE",
+            "Failed to install thread dump signal handler: " + t.getCause());
+      }
+    }
   }
 
```

Comment:
I think this is too late. Many a times we need this whole GemFireStore is starting up (due to hangs there), so the handler should be installed upfront. I will suggest setting the "isSnappyStore" flag too upfront via an internal property passed to GemFireStore.boot, or even a system property, or set a static field in GemFireStore instead of instance field so it can be set right at the start.